azure/azure_style.yaml

```diff
@@ -153,4 +156,25 @@ jobs:
           # Exit with an error if any of the tracked files changed
           git diff --summary --exit-code
 
+  - job: mypy
+    pool:
+      vmImage: "ubuntu-22.04"
+    timeoutInMinutes: ${{ parameters.TIMEOUT }}
+    continueOnError: ${{ parameters.IGNORE_STYLE }}
+    condition: ${{ parameters.MYPY }}
+    steps:
+      - checkout: self
+      - checkout: azure_template
+      - task: UsePythonVersion@0
+        inputs:
+          versionSpec: "3.11"
+      - script: |
+          cd ${{ parameters.REPO_NAME }}
+
+          pip install wheel
+          pip install mypy
+
+          # Run mypy check and print to screen
+          mypy .
```

Comment:
I think running like this does not work for two reasons:
1. This way, we also need to type tests. (I think this is excessive and not needed)
2. If your repo name is the same as your package name. (e.g. streamline/streamline) mypy will complain that the package 'streamline' already exists.

I have to run mypy like this: mypy streamline/

Reply:
You can set the files, modules and packages you want to type check in the `mypy.ini`: https://mypy.readthedocs.io/en/stable/config_file.html#confval-packages

But this only works if you invoke `mypy` without the dot. Thus we should change the above line to:

`mypy`